modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py

```diff
@@ -2139,6 +2139,43 @@ def test_shape_hint_detection_from_arrow(self):
         df = pd.utils.from_arrow(at)
         assert df._query_compiler._shape_hint == "column"
 
+    def test_constructor_from_modin_series(self):
+        def construct_has_common_projection(lib, df, **kwargs):
+            return lib.DataFrame({"col1": df.iloc[:, 0], "col2": df.iloc[:, 1]})
+
+        def construct_no_common_projection(lib, df1, df2, **kwargs):
+            return lib.DataFrame(
+                {"col1": df1.iloc[:, 0], "col2": df2.iloc[:, 0], "col3": df1.iloc[:, 1]}
+            )
+
+        def construct_mixed_data(lib, df1, df2, **kwargs):
+            return lib.DataFrame(
+                {
+                    "col1": df1.iloc[:, 0],
+                    "col2": df2.iloc[:, 0],
+                    "col3": df1.iloc[:, 1],
+                    "col4": np.arange(len(df1)),
+                }
+            )
+
+        run_and_compare(
+            construct_has_common_projection, data={"a": [1, 2, 3, 4], "b": [3, 4, 5, 6]}
+        )
+        run_and_compare(
+            construct_no_common_projection,
+            data={"a": [1, 2, 3, 4], "b": [3, 4, 5, 6]},
+            data2={"a": [10, 20, 30, 40]},
+            # HDK doesn't support concatenation of frames that has no common projection
+            force_lazy=False,
```

Comment:
@dchigarev Could you give some details on how this comment relates to `force_lazy` parameter?

Reply:
`force_lazy=True` means that the test is intended to fail if any non-lazy execution is triggered in the process. 

When Hdk doesn't support something it defaults to pandas which includes executing all of the delayed computation tree + materializing the table. This flow violates "force lazy" flag and thus results into an error.

When we specify `force_lazy=False` here we explicitly claim that this test actually defaults to pandas and only tests that the falling back process itself happens with no errors.

Reply:
@dchigarev thank you! It will be great, in the future, to have a docstring for `run_and_compare` utility, with a description of the parameters.